src/mon/PGMap.cc

```diff
@@ -3850,6 +3856,33 @@ static void _try_mark_pg_stale(
     newstat->state |= PG_STATE_STALE;
     newstat->last_unstale = ceph_clock_now();
   }
+
+    if ((cur.state & PG_STATE_PEERING) == 0 &&
```

Comment:
From what I understand, you're trying to update the last_peered state of the PG. However, this is already done in src/osd/PeeringState.cc. Therefore I think this block of code is not relevant.

Reply:
Will discard it and commit again.